pkg/controller/replicaset/replica_set.go

```diff
@@ -198,6 +203,17 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 // getPodReplicaSet returns the replica set managing the given pod.
 // TODO: Surface that we are ignoring multiple replica sets for a single pod.
 func (rsc *ReplicaSetController) getPodReplicaSet(pod *api.Pod) *extensions.ReplicaSet {
+	// look up in the cache, if cached and the cache is valid, just return cached value
+	// Note: if we create a new RS that is also matching a given pod, it will just be ignored
+	// (assuming that the previously matching RS didn't changed), but we sort overlapping RSs
+	// by creation timestamp, so it's no problem. If we change the sort order, we may need
+	// change the code here too.
```

Comment:
Hmmm... there is one issue with this.
Imagine that you have two RS:
- old RS1
- new RS2
You also have a pod that is attached to RS2 (because it doesn't match RS1 selector).

Now imagine that you are changing RS1 selector so that it is now matching that pod.
In the currently existing code, this would trigger the change for pod to be now owned by RS1.
With your cache this is no longer the case.

@bgrant0607 @lavalamp [so we are changing the semantics here - which may or may not be good - I'm still not sure].